File: AddStrings.py
# https://leetcode.com/problems/add-strings/description/
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        i,j=len(num1)-1,len(num2)-1
        carry=0
        res=[]
        while i>=0 or j>=0 or carry:
            dig1=int(num1[i]) if i>=0 else 0 #check if bothpointers are in bounds
            dig2=int(num2[j]) if j>=0 else 0
            total=dig1+dig2+carry
            carry=total//10
            res.append(str(total%10))
            i-=1
            j-=1
        return ''.join(res[::-1])

File: AddStrings.java
// https://leetcode.com/problems/add-strings/description/
class AddStrings{
    public String addStrings(String num1, String num2) {
        int i=num1.length()-1, j = num2.length()-1,carry=0,total;
        StringBuilder res= new StringBuilder();
        while(i>=0 || j>=0 || carry>0){
            int dig1= (i>=0?num1.charAt(i)-'0':0);
            int dig2= (j>=0?num2.charAt(j)-'0':0);
            total=dig1+dig2+carry;
            carry=total/10;
            res.append(total%10);
            i--;
            j--;
        }
        return res.reverse().toString();
    }
}